pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -42,12 +43,60 @@ type nodeConfig struct {
 	node *v1.Node
 }
 
+// discoverKubeAPIServerEndpoint discovers the kubernetes api server endpoint from the
+func discoverKubeAPIServerEndpoint() (string, error) {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return "", errors.Wrap(err, "unable to get config to talk to kubernetes api server")
+	}
+	if len(cfg.Host) == 0 {
+		return "", errors.Wrap(err, "could not get host name for the kubernetes api server")
+	}
+	return cfg.Host, nil
+}
+
 // NewNodeConfig creates a new instance of nodeConfig to be used by the caller.
-func NewNodeConfig(clientset *kubernetes.Clientset, windowsVM types.WindowsVM) *nodeConfig {
-	return &nodeConfig{k8sclientset: clientset, Windows: windows.New(windowsVM)}
+func NewNodeConfig(clientset *kubernetes.Clientset, windowsVM types.WindowsVM) (*nodeConfig, error) {
+	var workerIgnitionEndpoint string
+	var err error
+	if nodeConfigCache.workerIgnitionEndPoint == "" {
+		var kubeAPIServerEndpoint string
+		// We couldn't find it in cache. Let's compute it now.
```

Comment:
if this is sucessful shouldn't it be added to the cache?

Reply:
Ya, I missed that. I had it my previous commit. Thnx for catching it.

Reply:
There seems to be common code between `init()` and here. So a helper will be useful.

Reply:
Shouldn't we just fail in `init()` then? I would rather have the operator keep restarting till it can get the endpoint than failing during a scale up operation.